src/core/server/plugins/plugins_service.ts

```diff
@@ -256,4 +258,13 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
         )
     );
   }
+
+  private registerPluginStaticDirs(deps: PluginsServiceSetupDeps) {
+    for (const [pluginName, pluginInfo] of this.uiPluginInternalInfo) {
+      deps.http.registerStaticDir(
+        `/plugins/${pluginName}/assets/{path*}`,
+        pluginInfo.publicAssetsDir
+      );
```

Comment:
How should we document this feature?

Reply:
I added docs to the migration guide. Let me know if I can improve something.

Reply:
Is that a temporary measure to allow plugins to move their static assets to NP plugins until https://github.com/elastic/kibana/issues/50654 lands? Didn't we want to make all assets exposition explicit? Or is this going to be permanent to allow client-only plugins to expose assets?

Reply:
> Didn't we want to make all assets exposition explicit?

I'm on the fence here. On the one hand, we prefer explicitness, but it increases the API surface and makes a folder structure less predictable.
If we decided to add explicit API it should be something locked within the `plugin` folder. https://github.com/elastic/kibana/pull/60490#discussion_r395463220